cluster/mesos/cluster.go

```diff
@@ -72,10 +73,15 @@ func NewCluster(scheduler *scheduler.Scheduler, store *state.Store, TLSConfig *t
 	// Empty string is accepted by the scheduler.
 	user, _ := options.String("mesos.user", "SWARM_MESOS_USER")
 
+	hostname, _ := os.Hostname()
+	if hostname == "" {
+		hostname = frameworkName
```

Comment:
hmm, I think if we can't get the hostname and there are no overrides, I wonder if we can just don't pass override?
Passing framework name doesn't even seem logical to me..